chia/consensus/block_creation.py

```diff
@@ -166,7 +140,7 @@ def create_foliage(
 
         # Calculate the cost of transactions
         if new_block_gen is not None:
-            cost = compute_cost(new_block_gen, constants, height)
```

Comment:
this is where we save time. Instead of calling `compute_cost()`, we just use the cost field from `new_block_gen`.